src/samples/mavlink/goto.c

```diff
@@ -0,0 +1,230 @@
+        if (err < 0) {
+            SOL_ERR("Could not send vehicle to: (%f, %f, %f) - %s",
+                dest.latitude, dest.longitude, dest.altitude,
+                sol_util_strerrora(-err));
+            return;
+        }
+
+        printf(">>>> Successful takeoff, starting a new mission, heading to: "
+            "(%f, %f, %f)\n", dest.latitude, dest.longitude, dest.altitude);
+    }
+}
+
+static void
+mode_changed_cb(void *data, struct sol_mavlink *mavlink)
+{
+    int err;
+    sol_mavlink_mode mode = sol_mavlink_get_mode(mavlink);
+    bool armed = sol_mavlink_check_armed(mavlink);
+
+    if (mode == SOL_MAVLINK_MODE_GUIDED && !armed) {
+        err = sol_mavlink_set_armed(mavlink, true);
+        if (err < 0) {
+            SOL_ERR("Could not arm vechicle: %s", sol_util_strerrora(-err));
+        }
+    }
+}
+
+static void
+disarmed_cb(void *data, struct sol_mavlink *mavlink)
+{
+    sol_mavlink_mode mode;
+
+    mode = sol_mavlink_get_mode(mavlink);
+    if (mode == SOL_MAVLINK_MODE_LAND)
+        printf(">>>> Landed...\n");
+}
+
+static void
+armed_cb(void *data, struct sol_mavlink *mavlink)
+{
+    sol_mavlink_mode mode;
+
+    SOL_DBG("vehicle just armed");
+
+    mode = sol_mavlink_get_mode(mavlink);
+    if (mode == SOL_MAVLINK_MODE_GUIDED) {
+        takeoff(mavlink);
+    }
+}
+
+static void
+mavlink_connect_cb(void *data, struct sol_mavlink *mavlink)
+{
+    int err;
+    sol_mavlink_mode mode;
+
+    SOL_INF("mavlink connection stablished");
+
+    mode = sol_mavlink_get_mode(mavlink);
+    if (mode != SOL_MAVLINK_MODE_GUIDED) {
+        err = sol_mavlink_set_mode(mavlink, SOL_MAVLINK_MODE_GUIDED);
+        if (err < 0) {
+            SOL_ERR("Could not set mode: %s", sol_util_strerrora(-err));
+        }
+        return;
+    }
+
+    if (!sol_mavlink_check_armed(mavlink)) {
+        err = sol_mavlink_set_armed(mavlink, true);
+        if (err < 0) {
+            SOL_ERR("Could not arm vechicle: %s", sol_util_strerrora(-err));
+        }
+        return;
+    }
+
+    takeoff(mavlink);
+}
+
+int
+main(int argc, char *argv[])
+{
+    struct sol_mavlink *mavlink;
+    struct sol_mavlink_handlers mavlink_handlers = {
+        .connect = mavlink_connect_cb,
+        .mode_changed = mode_changed_cb,
+        .armed = armed_cb,
+        .disarmed = disarmed_cb,
+        .position_changed = position_changed_cb,
+        .mission_reached = mission_reached_cb,
+    };
+    struct sol_mavlink_config config = {
+        .handlers = &mavlink_handlers,
+    };
+
+    sol_init();
+
+    if (argc < 2) {
+        SOL_ERR("Usage: %s <address>", argv[0]);
+        goto err;
+    }
+
+    mavlink = sol_mavlink_connect(argv[1], &config, NULL);
+    if (!mavlink) {
+        SOL_ERR("Unable to stablish a Mavlink connection");
+        goto err;
+    }
+
+    sol_run();
+    sol_mavlink_disconnect(mavlink);
+    sol_shutdown();
+    return EXIT_SUCCESS;
+
+err:
+    sol_shutdown();
+    return EXIT_FAILURE;
+}
```

Comment:
lot's of code duplication between basic and goto samples

maybe you could have a common implementation?

also a description about these samples could be nice, making clear the difference between these samples